packages-exp/auth-exp/src/platform_cordova/popup_redirect/events.ts

```diff
@@ -41,6 +49,67 @@ export function _generateNewEvent(
   };
 }
 
+export function _savePartialEvent(auth: Auth, event: AuthEvent): Promise<void> {
+  return storage()._set(key(auth), (event as object) as PersistedBlob);
+}
+
+export async function _getAndRemoveEvent(
+  auth: Auth
+): Promise<AuthEvent | null> {
+  const event = (await storage()._get(key(auth))) as AuthEvent | null;
+  if (event) {
+    await storage()._remove(key(auth));
+  }
+  return event;
+}
+
+export function _eventFromPartialAndUrl(
+  partialEvent: AuthEvent,
+  url: string
+): AuthEvent | null {
+  // Parse the deep link within the dynamic link URL.
+  const callbackUrl = _getDeepLinkFromCallback(url);
+  // Confirm it is actually a callback URL.
+  // Currently the universal link will be of this format:
+  // https://<AUTH_DOMAIN>/__/auth/callback<OAUTH_RESPONSE>
+  // This is a fake URL but is not intended to take the user anywhere
+  // and just redirect to the app.
+  if (callbackUrl.includes('/__/auth/callback')) {
```

Comment:
This will have a false positive with `http://foo/#/__/auth/callback` (note the hash fragment). Considering using `new URL(callbackUrl).pathname` for proper URL parsing (with caveat of browser compatibility).

Reply:
This is copied more or less directly from the old SDK so I'm inclined to keep it: https://github.com/firebase/firebase-js-sdk/blob/master/packages/auth/src/cordovahandler.js#L707

This URL comes back from the handler so we can be confident there won't be this edge case in any case

Reply:
sg, let's keep it that way